src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -143,6 +146,26 @@ private void updateAssociatedTransaction(LoanDTO loan, Consumer<InnReachTransact
       .ifPresent(transactionConsumer);
   }
 
+  @Override
+  public void borrowerRenewLoan(LoanDTO loan) {
+    if (loan.getAction().equals("renewed")) {
+      var folioLoanId = loan.getId();
+
+      transactionRepository.fetchOneByLoanId(folioLoanId).ifPresent(transaction -> {
+        log.info("Loan {} associated with transaction ", folioLoanId);
+        var transactionDueDate = Instant.ofEpochSecond(transaction.getHold().getDueDateTime());
+        var loanDueDate = loan.getDueDate().toInstant().truncatedTo(ChronoUnit.SECONDS);
+        if (!loanDueDate.equals(transactionDueDate)) {
+          var loanIntegerDueDate = (int) (loan.getDueDate().getTime() / 1000);
+          String innReachRequestUri = resolveD2irCircPath("borrowerrenew", transaction.getTrackingId(), transaction.getCentralServerCode());
+          innReachExternalService.postInnReachApi(transaction.getCentralServerCode(), innReachRequestUri, loanIntegerDueDate);
```

Comment:
The payload must be set as dto/map. Also, take a look at `callD2irCircOperation`

Reply:
Done